exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -307,3 +315,246 @@ func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, clu
+		clusterv1.ClusterNameLabel:     mp.Spec.ClusterName,
+	}
+
+	if err := r.Client.List(ctx, machineList, client.InNamespace(mp.Namespace), client.MatchingLabels(labels)); err != nil {
+		return err
+	}
+
+	infraMachineNameToMachine := make(map[string]clusterv1.Machine)
+	for _, machine := range machineList.Items {
+		infraRef := machine.Spec.InfrastructureRef
+		infraMachineNameToMachine[infraRef.Name] = machine
+	}
+
+	for i := range infraMachines {
+		infraMachine := &infraMachines[i]
+		ownerRefs := infraMachine.GetOwnerReferences()
+		hasOwnerMachine := false
+		for _, ownerRef := range ownerRefs {
+			if ownerRef.Kind == "Machine" && ownerRef.APIVersion == clusterv1.GroupVersion.String() {
+				hasOwnerMachine = true
+				break
+			}
+		}
+
+		if !hasOwnerMachine {
+			machine, ok := infraMachineNameToMachine[infraMachine.GetName()]
+			if !ok {
+				return errors.Errorf("failed to patch ownerRef for infraMachine %q because no Machine has an infraRef pointing to it", infraMachine.GetName())
+			}
+			// Set the owner reference on the infraMachine to the Machine since the infraMachine is created and owned by the infraMachinePool.
+			infraMachine.SetOwnerReferences([]metav1.OwnerReference{
+				*metav1.NewControllerRef(&machine, machine.GroupVersionKind()),
+			})
+
+			patchHelper, err := patch.NewHelper(infraMachine, r.Client)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create patch helper for infraMachine %q in namespace %q", infraMachine.GetName(), infraMachine.GetNamespace())
+			}
+
+			if err := patchHelper.Patch(ctx, infraMachine); err != nil {
+				return errors.Wrapf(err, "failed to patch infraMachine %q in namespace %q", infraMachine.GetName(), infraMachine.GetNamespace())
+			}
+		}
+	}
+
+	return nil
+}
+
+// getNewMachine creates a new Machine object. The name of the newly created resource is going
+// to be created by the API server, we set the generateName field.
+func getNewMachine(mp *expv1.MachinePool, infraMachine *unstructured.Unstructured) *clusterv1.Machine {
+	infraRef := corev1.ObjectReference{}
+	if infraMachine != nil {
+		infraRef.APIVersion = infraMachine.GetAPIVersion()
+		infraRef.Kind = infraMachine.GetKind()
+		infraRef.Name = infraMachine.GetName()
+		infraRef.Namespace = infraMachine.GetNamespace()
+	}
+
+	annotations := mp.Spec.Template.Annotations
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	machine := &clusterv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-", mp.Name),
+			// Note: by setting the ownerRef on creation we signal to the Machine controller that this is not a stand-alone Machine.
+			OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(mp, mp.GroupVersionKind())},
+			Namespace:       mp.Namespace,
+			Labels:          make(map[string]string),
+			Annotations:     annotations,
+		},
+		Spec: clusterv1.MachineSpec{
+			ClusterName:       mp.Spec.ClusterName,
+			InfrastructureRef: infraRef,
```

Comment:
The drawback there would be that since all the MPM bootstrap refs point to the same object, deleting an MPM manually would not delete both its child resources like we'd normally expect.

Reply:
I think I would prefer not tracking the BootstrapConfig in the Machine for now over having to add branching to the reconcileDelete in the Machine controller

Reply:
I'm +1 on not doing this for now and exploring this in a separate issue/follow up since the PR scope is already quite large